src/MICore/CommandFactories/MICommandFactory.cs

```diff
@@ -54,6 +54,8 @@ public abstract class MICommandFactory
 
         public abstract string Name { get; }
 
+        internal string Version { get; set; }
```

Comment:
Can this be an int so we dont have to constantly run `int.TryParse(Version.Split('.').FirstOrDefault(), out int gdbMainVersion);`?

If its `null`, we could use majorVersion as -1 for invalid.

```suggestion
        internal int majorVersion { get; set; }
```

Reply:
I have changed the property to int.
Since the return of the method GdbVersionFromLog is a string, I had to parse it one time though.
Please let me know if this is alright!